File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
from heapq import *
def solution(scoville, K):
    heapify(scoville)
    counter = 0
    while scoville[0] < K:
        mix = heappop(scoville) + heappop(scoville) * 2
        heappush(scoville,mix)
        counter += 1
        if len(scoville) == 1 and scoville[0] < K:
            counter = -1
            break
    return counter

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] scoville, int K) {
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for(int i=0; i< scoville.length; i++){
            pq.add(scoville[i]);
        }
        
        
        int counter = 0;
        
        while(pq.peek() < K){

            if(pq.size() == 1){
                return -1;
            }

            int newScoville = pq.poll() + pq.poll() *2;
            pq.add(newScoville);
            counter++;

        }
        
        return counter;
    }
}